doc/classes/float.xml

```diff
@@ -33,7 +33,7 @@
 			<argument index="0" name="from" type="String">
 			</argument>
 			<description>
-				Cast a [String] value to a floating-point value. This method accepts float value strings like [code]"1.23"[/code] and exponential notation strings for its parameter so calling [code]float("1e3")[/code] will return 1000.0 and calling [code]float("1e-3")[/code] will return 0.001.
+				Cast a [String] value to a floating-point value. This method accepts float value strings like [code]"1.23"[/code] and exponential notation strings for its parameter so calling [code]float("1e3")[/code] will return 1000.0 and calling [code]float("1e-3")[/code] will return 0.001. Calling this method with an invalid float string will return 0. This method will ignore all non-number characters that can't be interpreted as the exponential notation, so calling [code]int("1a3")[/code] will return 13.
```

Comment:
"characters that can't be interpreted as the exponential notation" is a long way to write "letters that aren't e".

Reply:
I didn't test what would happen if you use 'e' outside of the exponential notation. For example, trying to convert with `float("1ea3")` will produce 13 or 1000? So to be safe I tried to be specific.

It's a good thing to have in the docs though. I'll test these specific cases and fix my PR accordingly.

Reply:
`print(float("1ea3"))` prints `1` :)

I think the logic is that it parses the string char by char from left to right, and stops at the first invalid characters (i.e. not a number, not `e`/`E` or `.`). So `"1ea3"` is `"1e<stop>"`, and since there's no valid integer after `"e"`, it ends as `1`.

`"1e3a1"` for example gives `1000`, so everything starting from the first invalid char is dropped.
`"1aaa1"` also gives `1` following the same logic.

Reply:
BTW you're referring to `int("1a3")` in the `float()` documentation :)

Reply:
Ops, fixing it. Thanks.